src/components/menu/menu.ts

```diff
@@ -233,18 +277,43 @@ export class Menu extends Component<Props, SpreadsheetChildEnv> {
    * If the given menu is not disabled, open it's submenu at the
    * correct position according to available surrounding space.
    */
-  openSubMenu(menu: Action, menuIndex: number, ev: MouseEvent) {
-    const parentMenuEl = ev.currentTarget as HTMLElement;
-    if (!parentMenuEl) return;
+  openSubMenu(parentMenuEl: HTMLElement, parentMenu: Action) {
+    if (!parentMenuEl || !parentMenu) return;
     const y = parentMenuEl.getBoundingClientRect().top;
-
     this.subMenu.position = {
       x: this.position.x + this.props.depth * MENU_WIDTH,
       y: y - (this.subMenu.scrollOffset || 0),
     };
-    this.subMenu.menuItems = menu.children(this.env);
+    this.subMenu.menuIndex = this.clickableMenuItems.findIndex(
+      (menuItem) => menuItem.id === parentMenu.id
+    );
+    this.subMenu.menuItems = parentMenu!.children(this.env);
     this.subMenu.isOpen = true;
-    this.subMenu.parentMenu = menu;
+    this.subMenu.parentMenu = parentMenu;
+  }
+
+  openSubMenuViaMouse(parentMenu: Action, ev: MouseEvent) {
+    const parentMenuEl = ev.currentTarget as HTMLElement;
+    this.openSubMenu(parentMenuEl, parentMenu);
+  }
+
+  openSubMenuViaKeyboard() {
+    if (
+      this.subMenu.menuIndex === undefined ||
+      this.menuItemRefs[this.subMenu.menuIndex] === undefined
+    ) {
+      this.backToParentRight();
+      return;
+    }
+    const parentMenu = this.clickableMenuItems[this.subMenu.menuIndex];
+    if (parentMenu.children(this.env).length === 0) {
+      this.backToParentRight();
+      return;
+    }
+    const { ref } = this.menuItemRefs[this.subMenu.menuIndex];
+    const parentMenuEl = ref.el as HTMLElement;
```

Comment:
the `el` is actually undefined once the component is unmounted, it would make sense to early return in that case